x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/analytics/action/AnalyticsStatsAction.java

```diff
@@ -97,109 +110,84 @@ protected void writeNodesTo(StreamOutput out, List<NodeResponse> nodes) throws I
-
-        public NodeResponse(DiscoveryNode node, long boxplotUsage, long cumulativeCardinalityUsage, long stringStatsUsage,
-                long topMetricsUsage, long ttestUsage) {
+    public static class NodeResponse extends BaseNodeResponse {
+        private final EnumCounters<Item> counters;
+
+        public NodeResponse(DiscoveryNode node, EnumCounters<Item> counters) {
             super(node);
-            this.boxplotUsage = boxplotUsage;
-            this.cumulativeCardinalityUsage = cumulativeCardinalityUsage;
-            this.stringStatsUsage = stringStatsUsage;
-            this.topMetricsUsage = topMetricsUsage;
-            this.ttestUsage = ttestUsage;
+            this.counters = counters;
         }
 
         public NodeResponse(StreamInput in) throws IOException {
             super(in);
-            if (in.getVersion().onOrAfter(Version.V_7_7_0)) {
-                boxplotUsage = in.readVLong();
-            } else {
-                boxplotUsage = 0;
-            }
-            cumulativeCardinalityUsage = in.readZLong();
-            if (in.getVersion().onOrAfter(Version.V_7_7_0)) {
-                stringStatsUsage = in.readVLong();
-                topMetricsUsage = in.readVLong();
-            } else {
-                topMetricsUsage = 0;
-                stringStatsUsage = 0;
-            }
             if (in.getVersion().onOrAfter(Version.V_7_8_0)) {
-                ttestUsage = in.readVLong();
+                counters = new EnumCounters<>(in, Item.class);
             } else {
-                ttestUsage = 0;
+                counters = new EnumCounters<>(Item.class);
+                if (in.getVersion().onOrAfter(Version.V_7_7_0)) {
+                    counters.inc(Item.BOXPLOT, in.readVLong());
+                }
+                counters.inc(Item.CUMULATIVE_CARDINALITY, in.readZLong());
+                if (in.getVersion().onOrAfter(Version.V_7_7_0)) {
+                    counters.inc(Item.STRING_STATS, in.readVLong());
+                    counters.inc(Item.TOP_METRICS, in.readVLong());
+                }
             }
         }
 
         @Override
         public void writeTo(StreamOutput out) throws IOException {
             super.writeTo(out);
-            if (out.getVersion().onOrAfter(Version.V_7_7_0)) {
-                out.writeVLong(boxplotUsage);
-            }
-            out.writeVLong(cumulativeCardinalityUsage);
-            if (out.getVersion().onOrAfter(Version.V_7_7_0)) {
-                out.writeVLong(stringStatsUsage);
-                out.writeVLong(topMetricsUsage);
-            }
             if (out.getVersion().onOrAfter(Version.V_7_8_0)) {
-                out.writeVLong(ttestUsage);
+                counters.writeTo(out);
+            } else {
+                if (out.getVersion().onOrAfter(Version.V_7_7_0)) {
+                    out.writeVLong(counters.get(Item.BOXPLOT));
+                }
+                out.writeZLong(counters.get(Item.CUMULATIVE_CARDINALITY));
```

Comment:
Should this be a VLong not ZLong?

Reply:
It was ZLong in 7.6, got partially changed by mistake in 7.7. So, I am returning to back to the original serialization method and I will open a PR to fix it in 7.7. Good catch, thanks for reminding me!

Reply:
I opened #55183. Not sure if it will make it into 7.7.0 or not.